starfish/util/logging.py

```diff
@@ -0,0 +1,58 @@
+import os
+import platform
+from functools import lru_cache
+from json import JSONEncoder
+from subprocess import (
+    CalledProcessError,
+    check_output,
+)
+from typing import Mapping
+
+import pkg_resources
+
+import starfish
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info() -> Mapping[str, str]:
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        version = get_dependency_version(dependency)
+        dependency_info[dependency] = version
+    return dependency_info
+
+
+def get_dependency_version(dependency: str) -> str:
+    return pkg_resources.get_distribution(dependency).version
+
+
+@lru_cache(maxsize=1)
+def get_git_commit_hash() -> str:
+    # First check if in starfish repo
+    os.chdir(os.path.dirname(starfish.__file__))
+    try:
+        check_output(["git", "ls-files", starfish.__file__])
+    except CalledProcessError:
+        return "Starfish project not under git tracking"
+    return check_output(["git", "describe", "--always"]).strip()
```

Comment:
@shanaxel42 : why did this get moved outside the try/except block? Won't this also throw if `ls-files` throws?

Reply:
well if `ls-files` throws then we just return the "Starfish project not under git tracking" string. So this should only be called if `ls-files` was valid

Reply:
🤦‍♂️ don't mind me.